### DISCUSSION TITLE
How do I install ansible-lint offline without using pip3?

### DISCUSSION THREAD

**Discussion by huitingyeong** [ansible/ansible-lint#2281] [General]
Hi I would like to install ansible-lint in a secure environment that does not have network connection and pip3. Is it possible?

**Comment by konstruktoid** [comment]
pip 19.0+ is required for installation (https://ansible-lint.readthedocs.io/en/latest/installing/#from-source) so that will be tricky.
otherwise download the source from e.g https://github.com/ansible/ansible-lint/releases/tag/v6.4.0 and install.

even in an air-gapped environment you will have to update and install things, so perhaps  request pip to be included in the allowed application list?

**Reply by ssbarnea** [reply]
There are lots of options:
- use pip to download all wheels and install from local cache, including pip itself
- use a pre-built container (like our creator-ee)
- use a system packages version (if your distro has it)
- host your own pypa registry (it is just a simple webserver)

**Reply by konstruktoid** [reply]
but w/o a network (I assume no internet access), those are tricky as well.

**Reply by huitingyeong** [reply]
Hi @konstruktoid How would I use pip3 to install the download tar.gz source file?

**Reply by konstruktoid** [reply]
`pip3 install <directory>`

But not having access to the Internet or a internal cache will cause a lot of work, since you still need to install any dependencies.

**Reply by huitingyeong** [reply]
Think I figured out my issue. Thought my installation command had a problem but it seems to be pip version is too low (9.X). Will try to figure that out. Thanks all for the kind guidance so far!
